### PR TITLE
Fix cppcheck warnings

### PR DESCRIPTION
**Pull request by eht16**
Is 26fd7328cd409ceb16557a0fe637d5c028c0598f enough for properly enabling the test program on `make check`? To me it seems, it was just missing or was it intentional that the test program isn't run on `make check`?

### CODE REVIEW

File: xmlsnippets/src/tests.c [hunk 1]
```diff
@@ -80,7 +80,9 @@ static gboolean test(gint ordinal, const gchar *input, const gchar *completion_n
 	else if (!completion_needed && !success)
 		return TRUE;
 
-	if (strcmp(c.completion, completion_needed) != 0)
+	if ((!completion_needed && c.completion) ||
+		(completion_needed && !c.completion) ||
+		strcmp(c.completion, completion_needed) != 0)
```
Comment (codebrainz) [inline xmlsnippets/src/tests.c]: Could also use [`g_strcmp0`](https://developer.gnome.org/glib/stable/glib-String-Utility-Functions.html#g-strcmp0).

Comment (eht16) [reply xmlsnippets/src/tests.c]: Of course, did it and re-pushed.

**Comment by eht16** [comment]
If there is nothing else, I'd like to merge this soon as the nightly builds are broken for days due to cppcheck issues.

**Comment by codebrainz** [comment]
LGBI

**Pull request merged**